### PR TITLE
allow nvme, since virtualbox 6.0 allows it

### LINKED ISSUE #8301: Virtualbox-iso - Add NVMe support option to hard_drive_interface?
It seems Virtualbox 6.0.xx series now support the NVMe controller. But when attempting to set hard_drive_interface to "NVMe", it says only IDE, Sata, or SCSI is supported. Can we add the option to allow the NVMe controller?

==> virtualbox-iso: Error creating disk controller: VBoxManage error: Oracle VM VirtualBox Command Line Management Interface Version 6.0.2
==> virtualbox-iso: (C) 2005-2019 Oracle Corporation
==> virtualbox-iso: All rights reserved.
==> virtualbox-iso:
==> virtualbox-iso: Usage:
==> virtualbox-iso:
==> virtualbox-iso: VBoxManage storagectl       <uuid|vmname>
==> virtualbox-iso:                             --name <name>
==> virtualbox-iso:                             [--add ide|sata|scsi|floppy|sas|usb|pcie]
==> virtualbox-iso:                             [--controller LSILogic|LSILogicSAS|BusLogic|
==> virtualbox-iso:                                           IntelAHCI|PIIX3|PIIX4|ICH6|I82078|
==> virtualbox-iso:                             [             USB|NVMe]

### PR DESCRIPTION
**Pull request by SwampDragons**
Enable NVMe controllers for pcie devices. 
Closes #8301

### CODE REVIEW

File: builder/virtualbox/iso/builder.go [hunk 1]
```diff
@@ -89,13 +89,19 @@ type Config struct {
 	// The type of controller that the primary hard drive is attached to,
 	// defaults to ide. When set to sata, the drive is attached to an AHCI SATA
 	// controller. When set to scsi, the drive is attached to an LsiLogic SCSI
+	// controller. When set to pcie, the drive is attached to an NVMe
```
Comment (azr) [inline builder/virtualbox/iso/builder.go]: This worked for me but I had to install an [extension pack](https://www.virtualbox.org/wiki/Downloads#VirtualBox6.0.14OracleVMVirtualBoxExtensionPack), may be we should document that too ?

**Comment by azr** [comment]
Ah wait; it seems that the controller is setup but disk is assigned to it : 
Screenshot 2019-11-01 at 12 36 03

File: builder/virtualbox/iso/step_create_disk.go [hunk 2]
```diff
@@ -73,6 +73,13 @@ func (s *stepCreateDisk) Run(ctx context.Context, state multistep.StateBag) mult
 			ui.Error(err.Error())
 			return multistep.ActionHalt
 		}
+	} else if config.HardDriveInterface == "pcie" {
+		if err := driver.CreateNVMeController(vmName, "NVMe Controller", config.NVMePortCount); err != nil {
```
Comment (azr) [inline builder/virtualbox/iso/step_create_disk.go]: Oh yes; I see I think this misses an 
```
if config.HardDriveInterface == "pcie" {
		controllerName = "NVMe Controller"
}
```

like line 94

Comment (azr) [reply builder/virtualbox/iso/step_create_disk.go]: Yes; I added this and it works !

**Comment by azr** [comment]
( got internal approval )

**Comment by breisig** [comment]
@azr I think we still have 1 issue here when virtualbox using 'pcie' mode is shutting down. When packer is trying to export  the VM, it shows the following error

```
==> virtualbox-iso: Preparing to export machine...
    virtualbox-iso: Deleting forwarded port mapping for the communicator (SSH, WinRM, etc) (host port 4429)
==> virtualbox-iso: Exporting virtual machine...
    virtualbox-iso: Executing: export centos-8-baseinstall --output source_images\centos-8-source\centos-8-baseinstall.ova
==> virtualbox-iso: Error exporting virtual machine: VBoxManage error: VBoxManage.exe: error: Cannot handle medium attachment: storageBus is 7, channel is 0, device is 0
==> virtualbox-iso: VBoxManage.exe: error: Details: code VBOX_E_NOT_SUPPORTED (0x80bb0009), component MachineWrap, interface IMachine, callee IUnknown
==> virtualbox-iso: VBoxManage.exe: error: Context: "ExportTo(pAppliance, Bstr(pszAbsFilePath).raw(), pVSD.asOutParam())" at line 1343 of file VBoxManageAppliance.cpp
==> virtualbox-iso: Provisioning step had errors: Running the cleanup provisioner, if present...
==> virtualbox-iso: Deregistering and deleting VM...
==> virtualbox-iso: Deleting output directory...
Build 'virtualbox-iso' errored: Error exporting virtual machine: VBoxManage error: VBoxManage.exe: error: Cannot handle medium attachment: storageBus is 7, channel is 0, device is 0
VBoxManage.exe: error: Details: code VBOX_E_NOT_SUPPORTED (0x80bb0009), component MachineWrap, interface IMachine, callee IUnknown
VBoxManage.exe: error: Context: "ExportTo(pAppliance, Bstr(pszAbsFilePath).raw(), pVSD.asOutParam())" at line 1343 of file VBoxManageAppliance.cpp

==> Some builds didn't complete successfully and had errors:
--> virtualbox-iso: Error exporting virtual machine: VBoxManage error: VBoxManage.exe: error: Cannot handle medium attachment: storageBus is 7, channel is 0, device is 0
VBoxManage.exe: error: Details: code VBOX_E_NOT_SUPPORTED (0x80bb0009), component MachineWrap, interface IMachine, callee IUnknown
VBoxManage.exe: error: Context: "ExportTo(pAppliance, Bstr(pszAbsFilePath).raw(), pVSD.asOutParam())" at line 1343 of file VBoxManageAppliance.cpp

==> Builds finished but no artifacts were created.
```
It seems every time packer tries to export, it always fails with this error. Of course using the following options.

```
      "hard_drive_interface": "pcie",
      "iso_interface": "sata",
      "vboxmanage": [
        [ "modifyvm", "{{.Name}}", "--firmware", "efi" ]
      ]
```

**Comment by azr** [comment]
Hey @breisig, yes I should not have merged that one; we resorted by marking this feature as beta and releasing it anyway. I think it needs more time too.. 🙂

**Pull request merged**